src/components/app/TabBar.js

```diff
@@ -51,6 +52,21 @@ class TabBar extends PureComponent {
   }
 }
 
+type Props = {
+  className?: string,
+  tabs: Tab[],
+  selectedTabName: string,
+  tabOrder: number[],
+  onSelectTab: string => mixed,
+  onChangeTabOrder: (number[]) => any,
+  +_mouseDownListener?: () => mixed,
```

Comment:
Will these three functions pass Flow with returning `void`? I usually prefer to be as strict as possible with the functions and how we use them. Also regarding your comment, I would make the signatures agree across components even if you need to modify another one to be consistent, and try to avoid the use of `mixed` and `any`.

I'm not sure the + is needed here on mouseDownListener. What's your reasoning for it?